### DISCUSSION TITLE
Why tokio named same as the capital of Japan?

### DISCUSSION THREAD

**Discussion by shaojintian** [tokio-rs/tokio#2783] [General] [answered]
Only a coincidence?

**Comment by Darksonn** [comment, marked as answer]
I don't know where Tokio's name comes from, but just to be clear, the capital of Japan is named Tokyo — not Tokio.

**Reply by shaojintian** [reply]
Thanks, it's my fault.

**Comment by hawkw** [comment]
I believe that the name originated as a pun on "Token IO", after the [`mio::Token`](https://docs.rs/mio/0.7.0/mio/struct.Token.html) type, since Tokio began its life as a layer of abstraction over `mio`.

**Reply by shaojintian** [reply]
I agree with you!
